packages/@react-spectrum/s2/stories/TableView.stories.tsx

```diff
@@ -1458,6 +1458,114 @@ let editableColumns: Array<Omit<ColumnProps, 'children'> & {name: string}> = [
+    let intermediateValue = useRef<any>(null);
+
+    let onChange = useCallback((id: Key, columnId: Key) => {
+      let value = intermediateValue.current;
+      if (value === null) {
+        return;
+      }
+      intermediateValue.current = null;
+      setEditableItems(prev => {
+        let newItems = prev.map(i => i.id === id && i[columnId] !== value ? {...i, [columnId]: value} : i);
+        return newItems;
+      });
+    }, []);
+
+    let onIntermediateChange = useCallback((value: any) => {
+      intermediateValue.current = value;
+    }, []);
+
+    return (
+      <div className={style({display: 'flex', flexDirection: 'column', gap: 16})}>
+        <TableView aria-label="Dynamic table" {...props} styles={style({width: 800, maxWidth: 'calc(100vw - 2rem)', height: 208})}>
+          <TableHeader columns={columns}>
+            {(column) => (
+              <Column {...column}>{column.name}</Column>
+            )}
+          </TableHeader>
+          <TableBody items={editableItems}>
+            {item => (
+              <Row id={item.id} columns={columns}>
+                {(column) => {
+                  if (column.id === 'fruits') {
+                    return (
+                      <EditableCell
+                        align={column.align}
+                        showDivider={column.showDivider}
+                        onSubmit={() => onChange(item.id, column.id!)}
+                        onCancel={() => {}}
+                        isSaving={item.isSaving[column.id!]}
+                        renderEditing={() => (
+                          <TextField
+                            aria-label="Edit fruit"
+                            autoFocus
+                            validate={value => value.length > 0 ? null : 'Fruit name is required'}
+                            styles={style({flexGrow: 1, flexShrink: 1, minWidth: 0})}
+                            defaultValue={item[column.id!]}
+                            onChange={value => onIntermediateChange(value)} />
+                        )}>
+                        <div className={style({display: 'flex', alignItems: 'center', gap: 8, justifyContent: 'space-between'})}>
+                          {item[column.id]}
+                          <ActionButton slot="edit" aria-label="Edit fruit">
+                            <Edit />
+                          </ActionButton></div>
+                      </EditableCell>
+                    );
+                  }
+                  if (column.id === 'farmer') {
+                    return (
+                      <EditableCell
+                        align={column.align}
+                        showDivider={column.showDivider}
+                        onSubmit={() => onChange(item.id, column.id!)}
+                        onCancel={() => {}}
+                        isSaving={item.isSaving[column.id!]}
+                        renderEditing={() => (
+                          <Picker
```

Comment:
just a nitpick, but I noticed that opening the picker here on mobile now properly expands the dropdown to grow to fit the available Table space, albiet with a slightly strange "grow" animation that only happens the first time the picker gets opened. However, I was wondering why it doesn't do that for the Async example? I was assuming it would use the width of the table as its bounds per say

Reply:
I also noticed that "grow" animation, I have no idea where it's coming from. There should be a max width on the popover of the table's width, https://github.com/adobe/react-spectrum/blob/43c7b0a91b77973c041c05219b28697b41188cbf/packages/%40react-spectrum/s2/src/TableView.tsx#L1211